traits.tests.ts

```diff
@@ -0,0 +1,873 @@
+import { ContractInterfaceFunction } from "@hirosystems/clarinet-sdk-wasm";
+import { initSimnet } from "@hirosystems/clarinet-sdk";
+import { readFileSync } from "fs";
+import { resolve } from "path";
+import {
+  buildTraitReferenceMap,
+  enrichInterfaceWithTraitData,
+  getContractIdsImplementingTrait,
+  isTraitReferenceFunction,
+} from "./traits";
+
+describe("Trait reference processing", () => {
+  it("correctly builds the trait reference map for a direct trait that is the first parameter", () => {
+    // Arrange
+    const allFunctionsInterfaces = JSON.parse(
+      readFileSync("./fixtures/direct-trait-functions-interfaces.json", "utf-8")
+    );
+
+    const expectedTraitReferenceMap = new Map(
+      Object.entries({
+        "test-trait": { token: "trait_reference" },
+      })
+    );
+
+    // Act
+    const traitReferenceMap = buildTraitReferenceMap(allFunctionsInterfaces);
```

Comment:
```suggestion
    const actual = buildTraitReferenceMap(allFunctionsInterfaces);
```

Reply:
How about using some generic naming here? It'll make the rest of the tests easier to follow when reading them after.